### ISSUE TITLE
Attribute require feature

### ISSUE THREAD

**Issue by Legion112** [vimeo/psalm#9879]
I would like to declare a class which implement my interface would be force to put annotation on class level.

Example
```php
#[\Attribute(\Attribute::TARGET_CLASS)]
class Meta {
}

/**
* @psalm-require-attribute Meta
*/
interface A {
}

```

In case of any class which implements my interface does not have Attribute Meta should report an error.

And logically if I read that annotation for any implementation of A interface attribute:
```
#[Meta]
class B implement A {}
       $notification = new B;
        $reflectionClass = new \ReflectionClass($notification);
        $attributes = $reflectionClass->getAttributes(Meta::class);

        /** Psalm should know that there is `Meta` attribute on instance A interface */
        $reflectionAttribute = $attributes[0];
        $attributeSetting = $reflectionAttribute->newInstance();
```
